config/database.yml

```diff
@@ -10,7 +10,7 @@ default: &default
   pool: 5
   username: <%= ENV['DATABASE_USER'] || "rr_dev" %>
   password: <%= ENV['DATABASE_PASS'] || "dev" %>
-  host: 127.0.0.1
+  host: localhost
```

Comment:
Looks like the test pipeline is failing due to this change from `127.0.0.1` to `localhost` can you either:
1. Make this accept an environment variable like how we do for `username` and `password` appropriately set it for the test pipeline
or
2. Return this to its original value?

Reply:
I made that change because of:
```
Mysql2::Error::ConnectionError:
  Can't connect to MySQL server on '127.0.0.1' (61)
```
I'm not sure why that is, since /etc/hosts just points localhost to the above address? But anyway, I changed this back to 127.0.0.1 in the `default` key, and overrode it for `test` and `development`.

Reply:
Duh, that wasn't going to work, since `test` is used both locally and in CI.
The root cause is that my install of MySQL 5.7 defaulted to use a domain socket in `/tmp/mysql.sock`. So I changed the code to dynamically choose `localhost` if that domain socket file exists; otherwise it uses 127.0.0.1.